pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/MetadataStoreException.java

```diff
@@ -156,6 +156,23 @@ public InvalidPathException(String path) {
         }
     }
 
+    /**
+     * Metadata operation timed out.
+     */
+    public static class TimeoutException extends RuntimeException {
```

Comment:
Why is a new exception needed? Should this extend MetadataStoreException? If not, why not use the ordinary TimeoutException?

Reply:
>If not, why not use the ordinary TimeoutException?

` java.util.concurrent.TimeoutException` is a checked exception. `BookieRackAffinityMapping::setConf` API belongs to different project (`BookKeeper`) and we can not throw any new random checked exception. Therefore, we need meaningful RuntimeException for metadata-Timeout. Therefore we added this TimeoutException into `MetadataStoreException` but `MetadataStoreException` is not a RuntimeException so, this new exception doesn't extend `MetadataStoreException`.